src/lib/viz/legend/LegendWidget.ts

```diff
@@ -0,0 +1,57 @@
+import { Layer } from '@/viz/layer';
+import { queryDOMElement } from '@/core/utils/dom';
+import { CartoError } from '@/core/errors/CartoError';
+
+export type LegendGeometryType = 'point' | 'line' | 'polygon';
+
+export type LegendProperties = {
+  type: LegendGeometryType;
+  color?: string;
+  marker?: string;
+  strokeColor?: string;
+  strokeStyle?: string;
+  label?: string;
+  width?: number;
```

Comment:
Why are we using here 'width' instead of 'size' (as in the style)? Should we include also other style properties like 'opacity'? 

I'm also seeing that we have a block of 'layer-style-related' props and 'others' (more specific to the Legend itself). Maybe we should try to organize them in 2 levels, so LegendProperty could have internally a Style for overrides

Reply:
As I explain in the previous comment, LegendProperties are the list of the properties that are supported by airship legend widgets so this property should be `width` because `size` is not supported by airship legend widgets.

About opacity, this property doesn't exist in airship legend widgets but checking CartoFrames I have seen that they include the opacity like the fourth component of the RGBA color so I've changed this PR to include the opacity and I'm going to change in the others legends PRs.

I agree with you to re-organize `LegendProperty` in two levels but I would like to do a bit of research and do this in other PR that will affect all legends PRs